.github/workflows/ci.yml

```diff
@@ -621,6 +636,30 @@ jobs:
         run : |
           cabal run -- bittide-instances:unittests
 
+  bittide-instances-hdl-matrix:
+    name: bittide-instances synthesis matrix generation
+    runs-on: self-hosted
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Generate matrix
+        run: |
+          if [[ "${{ github.event_name }}" == "schedule" || $(.github/scripts/force_expensive_checks.sh) == "true" ]]; then
+            cp .github/synthesis/all.json checks.json
+          else
+            cp .github/synthesis/staging.json checks.json
```

Comment:
Doesn't this alwasy overwrite the ouput generated in line 287?

Reply:
Every job runs in a clean environment and in parallel. So `contranomy-hdl` and `bittide-instances-hdl-matrix` don't share any state/filesystem.

Reply:
Right.